src/PrestaShopBundle/Controller/Admin/CategoryController.php

```diff
@@ -60,25 +65,43 @@ public function addSimpleCategoryFormAction(Request $request)
         if ($form->isValid()) {
             $data = $form->getData();
 
-            $_POST = [
-                'submitAddcategory' => 1,
-                'name_1' => $data['category']['name'],
-                'id_parent' => $data['category']['id_parent'],
-                'link_rewrite_1' => $tools->link_rewrite($data['category']['name']),
-                'active' => 1,
-                'checkBoxShopAsso_category' => $currentIdShop ? [$currentIdShop => $currentIdShop] : $shopList,
+            $localizedName = [
+                $defaultLanguageId => $data['category']['name'],
             ];
 
-            $adminCategoryController = $this->get('prestashop.adapter.admin.controller.category')->getInstance();
-            if ($category = $adminCategoryController->processAdd()) {
-                $response->setData(['category' => $category]);
-            }
+            $command = new AddCategoryCommand(
+                $localizedName,
+                [$defaultLanguageId => $tools->linkRewrite($data['category']['name'])],
+                true,
+                (int) $data['category']['id_parent']
+            );
+
+            $command->setAssociatedShopIds($currentIdShop ? [$currentIdShop => $currentIdShop] : $shopList);
+
+            try {
+                /** @var CategoryId $categoryId */
+                $categoryId = $commandBus->handle($command);
+
+                if ($categoryId->getValue()) {
+                    $response->setData(
+                        [
+                            'category' => [
+                                'id' => $categoryId->getValue(),
+                                'id_parent' => $data['category']['id_parent'],
+                                'name' => $localizedName,
+                            ],
+                        ]
+                    );
 
-            if ($request->query->has('id_product')) {
-                $productAdapter = $this->get('prestashop.adapter.data_provider.product');
-                $product = $productAdapter->getProduct($request->query->get('id_product'));
-                $product->addToCategories($category->id);
-                $product->save();
+                    if ($request->query->has('id_product')) {
```

Comment:
Even better, it should be a separate command i think.

Reply:
Even better! Is it possible for a command to launch a sub command?

Reply:
IMO this whole controller should be removed/refactored, this legacy logic was there a long time :confused:

Reply:
@jolelievre refactoring this would increase the scope of this PR even more, I suggest to keep it as is for this PR, as it's not something that was created by this PR. Wdyt?

Reply:
Alright, but you're lucky we are one day from freeze ^^ I'm gonna create an issue about this to be sure not to forget

Reply:
not that I don't want to fix it, I'm just aware that we are close to feature freeze :slightly_smiling_face: I'll gladly refactor this into a command later on

Reply:
I created this issue https://github.com/PrestaShop/PrestaShop/issues/13097